The size2 function is only available on ELF platforms (v. lib/ext2fs/getsize.c)

--- origsrc/e2fsprogs-1.41.12/e2fsck/unix.c	2010-05-17 22:04:40.000000000 -0500
+++ src/e2fsprogs-1.41.12/e2fsck/unix.c	2010-06-24 01:33:13.703832900 -0500
@@ -1127,7 +1127,7 @@ failure:
 		__u32 blocksize = EXT2_BLOCK_SIZE(fs->super);
 		int need_restart = 0;
 
-		pctx.errcode = ext2fs_get_device_size2(ctx->filesystem_name,
+		pctx.errcode = ext2fs_get_device_size(ctx->filesystem_name,
 						       blocksize,
 						       &ctx->num_blocks);
 		/*
@@ -1141,7 +1141,7 @@ failure:
 			ext2fs_close(fs);
 			need_restart++;
 			pctx.errcode =
-				ext2fs_get_device_size2(ctx->filesystem_name,
+				ext2fs_get_device_size(ctx->filesystem_name,
 							blocksize,
 							&ctx->num_blocks);
 		}
--- origsrc/e2fsprogs-1.41.6/misc/mke2fs.c	2009-04-18 09:38:50.000000000 -0500
+++ src/e2fsprogs-1.41.6/misc/mke2fs.c	2009-06-21 21:29:34.537489000 -0500
@@ -1448,7 +1448,7 @@ static void PRS(int argc, char *argv[])
 			fs_param.s_log_block_size = 2;
 			blocksize = 4096;
 		}
-		retval = ext2fs_get_device_size2(device_name,
+		retval = ext2fs_get_device_size(device_name,
 				 EXT2_BLOCK_SIZE(&fs_param), &big_dev_size);
 		if (retval)
 			goto get_size_failure;
--- origsrc/e2fsprogs-1.41.14/resize/main.c	2010-12-22 09:49:20.000000000 -0600
+++ src/e2fsprogs-1.41.14/resize/main.c	2011-01-06 19:28:47.516756300 -0600
@@ -374,7 +374,7 @@ int main (int argc, char ** argv)
 	 * defaults and for making sure the new filesystem doesn't
 	 * exceed the partition size.
 	 */
-	retval = ext2fs_get_device_size2(device_name, fs->blocksize,
+	retval = ext2fs_get_device_size(device_name, fs->blocksize,
 					&max_size);
 	if (retval) {
 		com_err(program_name, retval,
